packages/build-utils/src/vite.config.mjs

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import typescript from '@rollup/plugin-typescript';
+import process from 'process';
+import { defineConfig } from 'vite';
+
+export default defineConfig(({ mode }) => {
+  return {
+    build: {
+      lib: {
+        entry: 'src/index.ts',
+        // Build ES Module as CommonJS versions.
+        formats: ['es', 'cjs'],
+        fileName: (format, entryName) => {
+          // Since package.json contains `type: "module"`, .js files will be interpreted as ES Modules.
+          // CommonJS modules must be distinguished with a .cjs extension.
+          const extension = { es: 'js', cjs: 'cjs' }[format];
+          // Base output directory on the the module format.
+          // ES Modules are written to lib/es, CommonJS to lib/cjs.
+          return `${format}/${entryName}.${extension}`;
+        },
+      },
+      // Leave minification to consumers.
+      minify: false,
+      // Base output directory. Output paths are based on this and build.fileName.
+      outDir: 'lib',
+      rollupOptions: {
+        // Do not bundle dependencies.
+        external: /node_modules/,
+        output: {
+          paths: (id) => {
+            // ES Modules require imports to be fully specified (including extensions).
+            // Thus, we need to specify the extension for the fp.js file imported from lodash.
+            // This allows us to omit it in source files as we've done in the past in Terra UI.
+            if (id === 'lodash/fp') {
```

Comment:
having to know about a likely growing list of exceptions like this make me super uncomfortable...

Reply:
Yeah, I was trying to see if this could be done without requiring changes to source code. Alternatively, we can enable the [`import/extensions`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md) ESLint rule in packages. That would also require adding extensions to relative imports.

Reply:
After this PR, I will probably open another that replaces this with the `import/extensions` rule.

Reply:
forcing codebase changes feels drastic.... lets noodle on this hack and see if a nicer option bubbles up.... in the meantime i think it's ok to live with this until n is greater then one on id check.

Reply:
There are a few other cases where we import a file from within a package.

```
clipboard-polyfill/text
date-fns/fp
highcharts/modules/accessibility
highcharts/modules/exporting
lodash/fp
rc-slider/lib/Slider
react-dom/client
react-select/async-creatable
```

Several of those (`clipboard-polyfill/text, `rc-slider/lib/Slider`, and `react-select/async-creatable`) will likely end up in this package, so we can revisit this then.